Feature/1314 add data element geographic coordinates

KHORCASH opened the pull request.

jacobkellas commented:
@KHORCASH let's see if we can wire this up to use the phone's GPS.  I believe you had a branch that did something similar to this a year ago or so.

jacobkellas commented:
AB#2379

KHORCASH commented:
See: https://github.com/SanDiegoCountySheriff/Cal-RIPA/pull/653/commits/ca9aa5c905e51401c2471432396f411974ca98ee

Latest commit adds:
- RipaLocation async function to check for geo availability score, disables 'Current Location' button if availability score is > 200 (brought over from branch 1570)
- moved Lat and Long input text fields out of Other location option section and is now under block# and street
- Added index.js validation of coordinates from getCurrentLocation function to match RipaLocation lat and long validation rules. (allow max of 3 numbers after the decimal point)
- Added unit test for Geo availability function

jacobkellas commented on UI/src/components/molecules/RipaLocation.vue:
```diff
@@ -288,9 +336,16 @@ export default {
-    isGeolocationAvailable() {
-      return navigator.geolocation
-    },
+    // isGeolocationAvailable() {
```
Remove commented code

jacobkellas commented on UI/src/components/molecules/RipaLocation.vue:
```diff
@@ -444,6 +538,9 @@ export default {
     }
+
+    this.geolocationAvailable = await this.isGeolocationAvailable()
+    console.log('geoavail:', this.geolocationAvailable)
```
Remove console.log

jacobkellas commented on UI/src/components/molecules/RipaLocation.vue:
```diff
@@ -517,6 +614,17 @@ export default {
+      return new Promise(resolve => {
+        navigator.geolocation.getCurrentPosition(position => {
+          const accuracyScore = position.coords.accuracy
+          console.log(accuracyScore)
```
Remove console.log

jacobkellas commented on UI/src/store/index.js:
```diff
@@ -622,13 +622,46 @@ export default new Vuex.Store({
+                  )
+                }
+
+                console.log(latDecimal)
```
Remove console.log

jacobkellas reviewed (changes requested):
Haven't pulled the branch yet to test, but just a few cosmetic changes.

jacobkellas commented on UI/src/components/molecules/RipaLocation.vue:
```diff
@@ -335,6 +390,42 @@ export default {
       ]
     },
 
+    latitudeRules() {
```
Lat and Long rules don't seem to be working.  I'm getting the error message after clicking "current location"

jacobkellas commented on UI/src/components/molecules/RipaLocation.vue:
```diff
@@ -133,6 +143,38 @@
+      <template v-if="this.model.stopVersion === 2">
+        <ripa-subheader text="-- or --"></ripa-subheader>
+
+        <v-row no-gutters>
```
Let's discuss if this should be at the top.  Also when I click current location, the "more location options" switch turns off and hides the field.

jacobkellas commented:
@KHORCASH actually this isn't quite done, I forgot about a couple bugs we found last week.  Let me see if I can compile a list and we'll go over this today.

jacobkellas reviewed (changes requested):
Bug fixes
